boa/code/converter.py

```diff
@@ -0,0 +1,224 @@
+
+            #            pdb.set_trace()
+            if pytoken.instruction.arg == Compare.GT:
+                tokenizer.convert1(VMOp.GT, pytoken)
+            elif pytoken.instruction.arg == Compare.GE:
+                tokenizer.convert1(VMOp.GTE, pytoken)
+            elif pytoken.instruction.arg == Compare.LT:
+                tokenizer.convert1(VMOp.LT, pytoken)
+            elif pytoken.instruction.arg == Compare.LE:
+                tokenizer.convert1(VMOp.LTE, pytoken)
+            elif pytoken.instruction.arg == Compare.EQ:
+                tokenizer.convert1(VMOp.EQUAL, pytoken)
+            elif pytoken.instruction.arg == Compare.IS:
+                tokenizer.convert1(VMOp.EQUAL, pytoken)
+            elif pytoken.instruction.arg in [Compare.NE, Compare.IS_NOT]:
+                tokenizer.convert1(VMOp.NUMNOTEQUAL, pytoken)
+            elif pytoken.instruction.arg == Compare.IN:
+                tokenizer.convert1(VMOp.SWAP, pytoken)
+                tokenizer.convert1(VMOp.HASKEY, pytoken)
+
+        # arrays
+        elif op == pyop.BUILD_LIST:
+            tokenizer.convert_new_array(pytoken)
+        elif op == pyop.DUP_TOP:
+            tokenizer.convert1(VMOp.DUP, pytoken)
+        elif op == pyop.YIELD_VALUE:
+            tokenizer.convert1(VMOp.REVERSE, pytoken)
+        elif op == pyop.STORE_SUBSCR:
+            tokenizer.convert_store_subscr(pytoken)
+        elif op == pyop.BINARY_SUBSCR:
+            tokenizer.convert1(VMOp.PICKITEM, pytoken)
+
+        # dict
+        elif op == pyop.BUILD_CONST_KEY_MAP:
+            tokenizer.convert1(VMOp.NEWMAP, pytoken)
+        elif op == pyop.BUILD_MAP:
+            tokenizer.convert1(VMOp.NEWMAP, pytoken)
+
+        elif op == pyop.BUILD_SLICE:
+            tokenizer.convert_build_slice(pytoken)
+
+        elif op in [pyop.CALL_FUNCTION, pyop.CALL_METHOD]:
+            tokenizer.convert_method_call(pytoken)
+        #        elif op == pyop.LOAD_METHOD:
+
+        elif op == pyop.POP_TOP:
+            pass
+            # if prev_token:
+            #     is_action = False
+            #     for item in tokenizer.method.module.actions:
+            #         if item.method_name == prev_token.func_name:
+            #             is_action = True
+            #
+            # if is_action:
+            #     tokenizer.convert1(VMOp.DROP, pytoken)
+
+        elif op == pyop.DUP_TOP_TWO:
+            tokenizer.convert_dup_top_two(pytoken)
+        elif op == pyop.ROT_THREE:
+            tokenizer.convert1(VMOp.ROT, pytoken)
+        elif op == pyop.ROT_TWO:
+            tokenizer.convert1(VMOp.SWAP)
+        elif op == pyop.RAISE_VARARGS:
+            pass
+        #        elif op == pyop.CALL_METHOD:
+        #            pass
+        elif op == pyop.EXTENDED_ARG:
+            tokenizer.convert1(VMOp.NOP, pytoken)
+        else:
+            raise ConverterException()
+
+
+class Neo3Converter(IConverter):
+    def pytoken_to_vm(self, tokenizer, pytoken, prev_token=None):
```

Comment:
when it comes time to implement this method there will probably be several snippets from version 2 that will be repeated for version 3 and we will want to avoid repeating code.